src/components/Navigation/components/Item/Item.tsx

```diff
@@ -57,7 +68,7 @@ enum MatchState {
   NoMatch,
 }
 
-export default class Item extends React.Component<Props, State> {
+export class BaseItem extends React.Component<CombinedProps, State> {
```

Comment:
I'm curious, why did you change the class to `BaseItem` and export like 
```tsx
export const Item = withAppProvider<Props>()(BaseItem);
export default Item
```
rather than 
```tsx
export default withAppProvider<Props>(Item)
````

Reply:
I needed to do this because we use `Item` to [create the sub-navigation](https://github.com/Shopify/polaris-react/blob/49294ee23209ec3a844e916f1a7af4cc524fc933/src/components/Navigation/components/Item/Item.tsx#L226) items and this also needs to be wrapped in `withAppProvider`.